ruby-specs/action.yml

```diff
@@ -0,0 +1,140 @@
+  github_token:
+    description: "GitHub token for API access. It is used to upload code coverage status."
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: Set environment variables
+      shell: bash
+      run: |
+        echo "RAILS_ENV=test" >> $GITHUB_ENV
+        echo "RACK_ENV=test" >> $GITHUB_ENV
+        echo "COVERAGE=true" >> $GITHUB_ENV
+
+    - uses: actions/checkout@v6
+
+    - uses: ruby/setup-ruby@v1
+      with:
+        bundler-cache: true
+
+    - name: Determine if MongoDB is needed
+      id: mongodb_check
+      shell: bash
+      run: ${GITHUB_ACTION_PATH}/scripts/mongodb-check.sh >> $GITHUB_OUTPUT
+      env:
+        WITH_MONGODB: "${{ inputs.mongodb }}"
+
+    - name: "Start MongoDB container"
+      if: ${{ steps.mongodb_check.outputs.should_run == 'true' }}
+      shell: bash
+      run: ${GITHUB_ACTION_PATH}/scripts/mongodb-start.sh
+
+    - name: Determine if Redis is needed
+      id: redis_check
+      shell: bash
+      run: ${GITHUB_ACTION_PATH}/scripts/redis-check.sh >> $GITHUB_OUTPUT
+      env:
+        WITH_REDIS: "${{ inputs.redis }}"
+
+    - name: "Start Redis container"
+      if: ${{ steps.redis_check.outputs.should_run == 'true' }}
+      shell: bash
+      run: ${GITHUB_ACTION_PATH}/scripts/redis-start.sh
+
+    - name: Prepare test database
+      shell: bash
+      run: |
+        if bundle exec rake --tasks | grep --quiet "db:test:prepare"; then
+          bundle exec rake db:test:prepare
+        fi
+        if bundle exec rake --tasks | grep --quiet "spec:prepare"; then
+          bundle exec rake spec:prepare
+        fi
+
+    - name: "Export .env.test environment variables into the GitHub Action environment"
+      shell: bash
+      run: test -f .env.test && grep -v '^#' .env.test >> "$GITHUB_ENV" || true
+
+    - name: Determine if Pact should be executed
+      id: pact_check
+      shell: bash
+      run: ${GITHUB_ACTION_PATH}/scripts/check-pact.sh >> $GITHUB_OUTPUT
+      env:
+        WITH_PACT: "${{ inputs.pact }}"
+
+    - name: Pact
+      if: ${{ steps.pact_check.outputs.should_run == 'true' }}
+      shell: bash
+      run: bundle exec rake pact:verify
+
+    - name: Run RSpec
+      shell: bash
+      run: bundle exec rspec
+
+    - name: Upload SimpleCov Report
+      uses: actions/upload-artifact@v5
+      with:
+        name: coverage-report
+        path: coverage/
+
+    - name: Generate Code Coverage Summary
+      shell: bash
+      run: |
+        if [ -f coverage/.last_run.json ]; then
+          LINE_COVERAGE=$(jq '.result.line' coverage/.last_run.json)
+          BRANCH_COVERAGE=$(jq '.result.branch' coverage/.last_run.json)
+          echo "## Code Coverage Summary" >> $GITHUB_STEP_SUMMARY
+          echo "Line Coverage: $LINE_COVERAGE%" >> $GITHUB_STEP_SUMMARY
+          echo "Branch Coverage: $BRANCH_COVERAGE%" >> $GITHUB_STEP_SUMMARY
+        else
+          echo "No coverage report found."
+        fi
+
+    - name: Set GitHub Status Check
```

Comment:
The coverage report as a commit status is working as expected in the api PR:  

<img width="1676" height="378" alt="Image" src="https://github.com/user-attachments/assets/66a9b814-9af5-42cd-a96e-28ea3d8a35ce" />

Reply:
Before merging the API PR, we will need UFS approval